modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java

```diff
@@ -1351,9 +1354,50 @@ public void testCreateZone() {
         assertEquals("test_profile", zone.storageProfiles().profiles().get(0).storageProfile());
     }
 
+    @Test
+    public void testSetDefaultZone() {
+        StorageProfileParams storageProfile = StorageProfileParams.builder()
+                .storageProfile("test_profile")
+                .build();
+
+        CatalogCommand createZoneCmd = CreateZoneCommand.builder()
+                .zoneName(TEST_ZONE_NAME)
+                .storageProfilesParams(List.of(storageProfile))
+                .build();
+
+        assertThat(manager.execute(createZoneCmd), willCompleteSuccessfully());
+
+        assertNotEquals(TEST_ZONE_NAME, defaultZone().name());
+
+        CatalogCommand setDefaultCmd = AlterZoneSetDefaultCatalogCommand.builder()
+                .zoneName(TEST_ZONE_NAME)
+                .build();
+
+        assertThat(manager.execute(setDefaultCmd), willCompleteSuccessfully());
+        assertEquals(TEST_ZONE_NAME, defaultZone().name());
+
+        assertThat(manager.execute(simpleTable(TABLE_NAME)), willCompleteSuccessfully());
+
+        Catalog catalog = Objects.requireNonNull(manager.catalog(manager.latestCatalogVersion()));
+
+        CatalogTableDescriptor tab = Objects.requireNonNull(manager.table(TABLE_NAME, catalog.time()));
```

Comment:
```suggestion
        CatalogTableDescriptor tab = Objects.requireNonNull(manager.table(TABLE_NAME, clock.nowLong()));
```

Reply:
Also, I'd suggest to check history wasn't changed by "set default zone" command.